common/geth/failover.go

```diff
@@ -0,0 +1,44 @@
+package geth
+
+import (
+	"sync"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+)
+
+type RPCStatistics struct {
```

Comment:
Not sure why you change to this name, this does more than statistics like counting num errors, but actually handle and make failover decisions. The old name may fit better.